kobo/apps/audit_log/tests/test_project_history_logs.py

```diff
@@ -862,3 +880,43 @@ def test_export_v1_creates_log(self):
         log = log_query.first()
         self._check_common_metadata(log.metadata, PROJECT_HISTORY_LOG_PROJECT_SUBTYPE)
         self.assertEqual(log.object_id, self.asset.id)
+
+    @data(
+        ('archive', AuditAction.ARCHIVE, False),
+        ('unarchive', AuditAction.UNARCHIVE, False),
+        ('undelete', AuditAction.UNDELETE, True),
+        ('delete', AuditAction.DELETE, True),
+    )
+    @unpack
+    def test_bulk_actions(self, bulk_action, audit_action, should_ignore):
+        assets = [Asset.objects.create(
+            content={
+                'survey': [
+                    {
+                        'type': 'text',
+                        'label': 'Question 1',
+                        'name': 'q1',
+                        '$kuid': 'abc',
+                    },
+                ]
+            },
+            owner=self.user,
+            asset_type='survey',
+        ) for i in range(0, 2)]
+
+        for asset in assets:
+            asset.deploy(backend='mock', active=True)
+
+        uids = [asset.uid for asset in assets]
+
+        if bulk_action == 'undelete':
+            self._make_bulk_request(uids, 'delete')
+
+        self._make_bulk_request(uids, bulk_action)
+        project_hist_logs = ProjectHistoryLog.objects.filter(
```

Comment:
Instead of filtering on action first (so we don't need AuditAction.UNDELETE), we can just say `if should_ignore: self.assertEqual(ProjectHistoryLog.objects.count(), 0)` at the top